src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -167,29 +167,49 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
                 name.Name = serializerName;
                 name.CodeBase = null;
                 name.CultureInfo = CultureInfo.InvariantCulture;
-                string serializerPath = Path.Combine(Path.GetDirectoryName(type.Assembly.Location), serializerName + ".dll");
-                if (!File.Exists(serializerPath))
+
+                string serializerPath = null;
+
+                try
+                {
+                    serializerPath = Path.Combine(Path.GetDirectoryName(type.Assembly.Location), serializerName + ".dll");
```

Comment:
If we know `type.Assembly.Location` can be null, we should take care of it instead of letting the exception bubble up.